app/assets/javascripts/cocoon.js

```diff
@@ -118,15 +118,16 @@
     }, timeout);
   });
 
-
-  $(document).on("ready page:load turbolinks:load", function() {
+  var hideRemoveFields = function() {
     $('.remove_fields.existing.destroyed').each(function(i, obj) {
       var $this = $(this),
           wrapper_class = $this.data('wrapper-class') || 'nested-fields';
 
       $this.closest('.' + wrapper_class).hide();
     });
-  });
+  }
+  $(hideRemoveFields); // On page load
+  $(document).on('page:load turbolinks:load', hideRemoveFields); // On custom load events
 
```

Comment:
Mmmmmm. I think this is not correct and I do not have a simple way to fix it. The `"ready page:load"` events are both replaced by the `"turbolinks:load"` event. So this works perfectly fine for backwards compatibility (if you use the new turbolinks). Indeed, if someone does _not_ use turbolinks it will no longer be triggered. But someone using turbolinks 5 will, with your code, call `hideRemoveFields` twice.